network/p2p/GENERAL_RULES.md

```diff
@@ -0,0 +1,69 @@
+# General message validations
+
+## Message reactions
+
+Node can react to a message in three ways:
+- `Accept` - message is valid, accepted and processed further.
+- `Ignore` - message is invalid, ignored and not processed further, violation count is increased.
+- `Reject` - message is rejected, ignored and not processed further, violation count is set to violation threshold, peer is banned for the current round.
+
+## Semantic assertions
+
+- Message validator maintains a state for each validator.
+- Generally, if violation happens by a small margin, the message is ignored. 
+  - If an assertion has a rule for ignoring, the rule has higher priority than this one.
+- Generally, If violation happens by a large margin, the message is rejected. 
+  - If an assertion has a rule for rejecting, the rule has higher priority than this one.
+- If violation count reaches a certain threshold, all further messages from the validator for current round are rejected.
+- Each round resets violation count to 0.
```

Comment:
Handle case if node is a bit malicious (ignored) each round but does it repetitively: should it be punished?